src/net/atlas/download.rs

```diff
@@ -43,6 +65,29 @@ impl AttachmentsDownloader {
         }
     }
 
+    pub fn has_ready_batches(&self) -> bool {
+        for batch in self.priority_queue.iter() {
```

Comment:
I think we need to have some kind of symmetry between this method, traversing the priority queue and, the method  `pop_next_ready_batch` just `peek`ing.
With this approach, you could end up returning `None`, which would result in a panic because of the expect, line 137.

Reply:
I'm not sure I follow? The previous code used `priority_queue.is_empty()` followed by a `priority_queue.pop().expect(..)`.  If anything, I'm adhering to the semantics that were already there.

Reply:
I could be missing something, but I think that this new conditions are equivalent to writing:
```
queue = [1, 1, 0]
if queue.contains(1) {
    assert!(queue.last() == 1);
}
```

Reply:
I changed the implementation of `Ord` for `AttachmentsBatch` so that the "first" `AttachmentsBatch` is the one with the earliest deadline.  If `has_ready_batches()` returns `true`, then not only is `self.priority_queue.len() > 0`, but also `self.priority_queue.front().as_ref().unwrap().retry_deadline < get_epoch_time_secs()`.

Reply:
could you add a comment for the fn `pop_next_ready_batch` explaining this?